### PR TITLE
Improve Knative

### LINKED ISSUE #179: Support CloudEvents v1.0.0

### LINKED ISSUE #326: Improve knative broker integration
When doing something like:

```
from("knative:event/x")
  .transform()
  .to("knative:event/y")
```

Then the event is delivered to the same service again. The reason is that the header containing the event type (x) overrides the event type "y" provided in the producer endpoint.

It's not so clear what happens.. Maybe we should make such override not enabled by default.

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #326 
Fixes #179

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@nicolaferraro please test it :)

**Comment by lburgazzoli** [comment]
> It's not clear how to make a dynamic event dispatch now. The event type in the endpoint is not overridden now, but what if I want it to be dynamic?

At the moment the solution would be to use `choice / when`.

Support for `knative:event` need to be evaluate but in order to support tooling to properly generate information about what events types are produced/consumed, having it explicit could help.

**Comment by nicolaferraro** [comment]
Last commit adds support for the `knative:event` uri. I'm adding the required changes in the operator.

File: camel-knative/camel-knative/src/main/java/org/apache/camel/component/knative/KnativeConfiguration.java [hunk 1]
```diff
@@ -31,8 +31,8 @@
     private KnativeEnvironment environment;
     @UriParam
     private String serviceName;
-    @UriParam(defaultValue = "0.3", enums = "0.1,0.2,0.3")
-    private String cloudEventsSpecVersion = CloudEvents.V03.version();
+    @UriParam(defaultValue = "0.1", enums = "0.1,0.2,0.3,1.0")
```
Comment (davsclaus) [inline camel-knative/camel-knative/src/main/java/org/apache/camel/component/knative/KnativeConfiguration.java:34]: Is this correct should default value not be "1.0" ?

Comment (lburgazzoli) [reply camel-knative/camel-knative/src/main/java/org/apache/camel/component/knative/KnativeConfiguration.java:34]: damn, will fix it

Comment (lburgazzoli) [reply camel-knative/camel-knative/src/main/java/org/apache/camel/component/knative/KnativeConfiguration.java:34]: done

**Pull request merged**